client/test/e2e.test.js

```diff
@@ -0,0 +1,131 @@
+import test from 'tape';
+import puppeteer from 'puppeteer';
+
+const width = 1920;
+const height = 1080;
+
+// Opens the chromium window
+function browser(width, height) {
+  return puppeteer.launch({
+    headless: true,
+    args: [`--window-size=${width},${height}`],
+    // slowMo: 20 // For testing
+  });
+}
+
+function sleep(ms) {
+  return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+// Clicks the add workspace button, types a name, and clicks the button
+async function create_workspace(p, name) {
+  await p.waitForSelector('#add-workspace');
+  await p.click('#add-workspace');
+  await p.waitForSelector('#workspace-name');
+  await p.focus('#workspace-name');
+  await p.keyboard.type(name);
+  await p.click('#create-workspace');
+}
+
+async function workspace_exists(p, name) {
+  let exists = false;
+
+  await p.waitForSelector('.v-list-item__title');
+  let workspaces = await p.evaluate(() => {
+    let titles = [];
+    let doc_nodes = document.querySelectorAll('.v-list-item__title');
+    for (let node of doc_nodes) {
+      titles.push(node.innerText);
+    }
+    return titles;
+  });
+
+  exists = workspaces.includes(name);
+
+  return exists;
+}
+
+async function element_exists(empty = false, element_type, p, name) {
+  let exists, list_is_empty, tables;
+
+  await p.waitForSelector('[data-title="Tables"] .ws-detail-empty-list ');
```

Comment:
```suggestion
  await p.waitForSelector(`[data-title="${element_type}"] .ws-detail-empty-list`);
```

This should use the `element_type` argument right?

Reply:
Yes, good catch

Reply:
Fixed in b3f9d38.